providers/aws/resources/aws_elasticache.go

```diff
@@ -219,3 +216,104 @@ func (a *mqlAwsElasticache) getCacheClusters(conn *connection.AwsConnection) []*
 	}
 	return tasks
 }
+
+func (a *mqlAwsElasticache) serverlessCaches() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	res := []interface{}{}
+	poolOfJobs := jobpool.CreatePool(a.getServerlessCaches(conn), 5)
+	poolOfJobs.Run()
+
+	// check for errors
+	if poolOfJobs.HasErrors() {
+		return nil, poolOfJobs.GetErrors()
+	}
+	// get all the results
+	for i := range poolOfJobs.Jobs {
+		if poolOfJobs.Jobs[i].Result != nil {
+			res = append(res, poolOfJobs.Jobs[i].Result.([]interface{})...)
+		}
+	}
+
+	return res, nil
+}
+
+func (a *mqlAwsElasticache) getServerlessCaches(conn *connection.AwsConnection) []*jobpool.Job {
+	tasks := make([]*jobpool.Job, 0)
+	regions, err := conn.Regions()
+	if err != nil {
+		return []*jobpool.Job{{Err: err}}
+	}
+
+	for _, region := range regions {
+		regionVal := region
+		f := func() (jobpool.JobResult, error) {
+			log.Debug().Msgf("elasticache>getServerlessClusters>calling aws with region %s", regionVal)
+
+			svc := conn.Elasticache(regionVal)
+			ctx := context.Background()
+			res := []interface{}{}
+
+			var marker *string
+			for {
+				caches, err := svc.DescribeServerlessCaches(ctx, &elasticache.DescribeServerlessCachesInput{
+					NextToken: marker,
+				})
+				if err != nil {
+					if Is400AccessDeniedError(err) {
+						log.Warn().Str("region", regionVal).Msg("error accessing region for AWS API")
+						return res, nil
+					}
+					return nil, err
+				}
+				if len(caches.ServerlessCaches) == 0 {
+					return nil, nil
+				}
+				for i := range caches.ServerlessCaches {
+					cache := caches.ServerlessCaches[i]
+
+					sgs := []interface{}{}
+					for i := range cache.SecurityGroupIds {
+						sgId := cache.SecurityGroupIds[i]
+						mqlSg, err := NewResource(a.MqlRuntime, "aws.ec2.securitygroup",
+							map[string]*llx.RawData{
+								"arn": llx.StringData(fmt.Sprintf(securityGroupArnPattern, regionVal, conn.AccountId(), sgId)),
```

Comment:
nit: i think it might be better if we have a `func generateSecurityGroupId(params....)` instead of using a regex

Reply:
extracted and moved to aws.go